docs/k8s-quickstart.asciidoc

```diff
@@ -224,20 +224,20 @@ Use `kubectl port-forward` to access Kibana from your local workstation:
 kubectl port-forward service/quickstart-kb-http 5601
 ----
 +
-Open `https://localhost:5601` in your browser. Your browser will show a warning because the self-signed certificate configured by default is not verified by a third party certificate authority and not trusted by your browser. You can either configure a link:k8s-accessing-elastic-services.html#k8s-setting-up-your-own-certificate[valid certificate] or acknowledge the warning for the purposes of this quick start.
+Open `https://localhost:5601` in your browser. Your browser will show a warning because the self-signed certificate configured by default is not verified by a third party certificate authority and not trusted by your browser. You can temporarily acknowledge the warning for the purposes of this quick start but it is highly recommended that you <<{p}-setting-up-your-own-certificate,configure valid certificates>> for any production deployments.
 +
-Login with the `elastic` user. Retrieve its password with:
+Login as the `elastic` user. The password can be obtained with the following command:
 +
 [source,sh]
 ----
-echo $(kubectl get secret quickstart-es-elastic-user -o=jsonpath='{.data.elastic}' | base64 --decode)
+kubectl get secret quickstart-es-elastic-user -o=jsonpath='{.data.elastic}' | base64 --decode; echo
 ----
 
 [float]
 [id="{p}-upgrade-deployment"]
 === Upgrade your deployment
 
-You can apply any modification to the original cluster specification. The operator makes sure that your changes are applied to the existing cluster, while avoiding downtime.
+You can add and modify most elements of the original cluster specification provided that they translate to valid transformations of the underlying Kubernetes resources (e.g., existing volume claims cannot be resized). The operator will attempt to apply your changes with minimal disruption to the existing cluster but it might not be possible to apply some changes gracefully. For complex topology changes, you should break your changes down to smaller incremental steps that the operatator can apply safely. You should also ensure that the Kubernetes cluster has sufficient resources to accommodate the changes (extra storage space, sufficient memory and CPU resources to temporarily spin up new pods etc.).
```

Comment:
> The operator will attempt to apply your changes with minimal disruption to the existing cluster but it might not be possible to apply some changes gracefully. For complex topology changes, you should break your changes down to smaller incremental steps that the operatator can apply safely.

That's quite scary. What kind of change do you have in mind that the operator is not allowed to handle smoothly? (1-node clusters and indices with 0 replicas have their own section in https://github.com/elastic/cloud-on-k8s/pull/1859, but that limitation seems just "normal").

Reply:
Well, my intention was to emphasise that users shouldn't expect miracles from the operator. For example, someone might try to shrink 10 full nodes down to 1 intentionally (or due to a typo). Even if the operator works correctly, I doubt that the final state of the cluster would be in a good state -- which is ultimately what the users want.

It was a polite way of saying "don't do stupid things" but if it comes across as us not being confident about the operator, I am happy to reconsider the wording.

Reply:
> For example, someone might try to shrink 10 full nodes down to 1 intentionally (or due to a typo). Even if the operator works correctly, I doubt that the final state of the cluster would be in a good state

Hm if we scale down from 10 to 1 no disruption should be expected. If one node is not enough to hold all data from the 10 nodes, the scale down will never finish, but the cluster should stay green the whole time.
Read/write performance is of course impacted but it seems unrelated to ECK as completely driven by the user.

I think I would keep it simple, from:

> The operator will attempt to apply your changes with minimal disruption to the existing cluster but it might not be possible to apply some changes gracefully. For complex topology changes, you should break your changes down to smaller incremental steps that the operatator can apply safely.

to:

> The operator will attempt to apply your changes with minimal disruption to the existing cluster.